lib/Wasmtime.hs

```diff
@@ -997,13 +998,13 @@ newtype MemoryType = MemoryType {unMemoryType :: ForeignPtr C'wasm_memorytype_t}
 
 -- | Creates a descriptor for a WebAssembly 'Memory' with the specified minimum number of memory pages,
 -- an optional maximum of memory pages, and a 64 bit flag, where false defaults to 32 bit memory.
-newMemoryType :: 
+newMemoryType ::
   -- | Minimum number of memory pages.
-  Word64 -> 
+  Word64 ->
   -- | Optional maximum of memory pages.
-  Maybe Word64 -> 
-   -- | 'True' means memory is 64 bit flag and 'False' means 32 bit.
-  Bool -> 
+  Maybe Word64 ->
+  -- | 'True' means memory is 64 bit flag and 'False' means 32 bit.
```

Comment:
btw, would it make sense to have an ADT for this instead of true/false?

Reply:
I think so. In general, we need to do pass through this C-adapted API and use more haskell idioms in the exposed functions. Thanks for pointing this out!